python/pyarrow/parquet.py

```diff
@@ -1891,6 +1892,14 @@ def write_to_dataset(table, root_path, partition_cols=None,
         file metadata instances of dataset pieces. The file paths in the
         ColumnChunkMetaData will be set relative to `root_path`.
     """
+    if use_legacy_dataset is None:
```

Comment:
What happens if the user specifies `True` and supplies a new filesystem?

Reply:
They will get an error about an unrecognized file system (the same error as you get now passing such a filesystem, as in ARROW-12299).

I could add a check for it and raise a bit more informative error message (but it should be rather rare to do that I think)